File: Presentation/Styling/SyntaxHighlighter.swift
import Cocoa

/// Syntax highlighter for code blocks using One Dark theme colors
class SyntaxHighlighter {

    // MARK: - One Dark Theme

    private let keyword = NSColor(red: 0.78, green: 0.47, blue: 0.80, alpha: 1.0)    // Purple
    private let type = NSColor(red: 0.90, green: 0.73, blue: 0.46, alpha: 1.0)       // Yellow
    private let string = NSColor(red: 0.59, green: 0.78, blue: 0.50, alpha: 1.0)     // Green
    private let number = NSColor(red: 0.82, green: 0.58, blue: 0.46, alpha: 1.0)     // Orange
    private let comment = NSColor(red: 0.45, green: 0.50, blue: 0.55, alpha: 1.0)    // Gray
    private let function = NSColor(red: 0.38, green: 0.71, blue: 0.93, alpha: 1.0)   // Blue
    private let plain = NSColor(red: 0.87, green: 0.89, blue: 0.91, alpha: 1.0)      // Off-white
    private let decorator = NSColor(red: 0.90, green: 0.73, blue: 0.46, alpha: 1.0)  // Yellow

    private let codeFont: NSFont
    private let codeBgColor = NSColor(red: 0.12, green: 0.13, blue: 0.15, alpha: 1.0)

    // Language keywords
    private let keywords: [String: Set<String>] = [
        "python": Set(["def", "class", "if", "elif", "else", "for", "while", "return", "import", "from",
                       "as", "try", "except", "finally", "raise", "with", "yield", "lambda", "pass",
                       "break", "continue", "and", "or", "not", "in", "is", "async", "await", "None",
                       "True", "False", "global", "nonlocal", "assert", "del"]),
        "java": Set(["public", "private", "protected", "class", "interface", "extends", "implements",
                     "static", "final", "void", "int", "long", "double", "float", "boolean", "char",
                     "byte", "short", "new", "return", "if", "else", "for", "while", "do", "switch",
                     "case", "default", "break", "continue", "try", "catch", "finally", "throw",
                     "throws", "import", "package", "this", "super", "null", "true", "false",
                     "abstract", "synchronized", "volatile", "instanceof", "enum", "var", "record"]),
        "javascript": Set(["function", "const", "let", "var", "if", "else", "for", "while", "do",
                           "switch", "case", "default", "break", "continue", "return", "try", "catch",
                           "finally", "throw", "class", "extends", "new", "this", "super", "import",
                           "export", "from", "as", "async", "await", "yield", "of", "in", "typeof",
                           "instanceof", "delete", "void", "null", "undefined", "true", "false", "NaN"]),
        "sql": Set(["SELECT", "FROM", "WHERE", "JOIN", "LEFT", "RIGHT", "INNER", "OUTER", "ON", "AND",
                    "OR", "NOT", "IN", "IS", "NULL", "LIKE", "BETWEEN", "GROUP", "BY", "ORDER", "ASC",
                    "DESC", "LIMIT", "OFFSET", "HAVING", "INSERT", "INTO", "VALUES", "UPDATE", "SET",
                    "DELETE", "CREATE", "TABLE", "ALTER", "DROP", "INDEX", "VIEW", "DISTINCT", "AS",
                    "CASE", "WHEN", "THEN", "ELSE", "END", "UNION", "ALL", "EXISTS", "COUNT", "SUM",
                    "AVG", "MAX", "MIN", "COALESCE", "CAST", "select", "from", "where", "join", "and",
                    "or", "not", "in", "is", "null", "like", "order", "by", "group", "having", "as"]),
        "swift": Set(["func", "class", "struct", "enum", "protocol", "extension", "var", "let", "if",
                      "else", "guard", "for", "while", "repeat", "switch", "case", "default", "break",
                      "continue", "return", "try", "catch", "throw", "throws", "defer", "import",
                      "self", "Self", "super", "nil", "true", "false", "public", "private", "internal",
                      "static", "final", "override", "mutating", "lazy", "weak", "async", "await"]),
        "go": Set(["func", "package", "import", "type", "struct", "interface", "var", "const", "if",
                   "else", "for", "range", "switch", "case", "default", "break", "continue", "return",
                   "go", "select", "chan", "map", "make", "new", "defer", "panic", "recover", "nil",
                   "true", "false", "iota"])
    ]

    private let builtinTypes = Set(["String", "Int", "Integer", "Long", "Double", "Float", "Boolean",
                                     "Bool", "List", "Array", "Dict", "Dictionary", "Map", "Set",
                                     "HashMap", "HashSet", "ArrayList", "Optional", "Result", "Void",
                                     "Object", "Number", "Date", "Error", "Exception", "Promise"])

    init(fontSize: CGFloat = 12) {
        self.codeFont = NSFont.monospacedSystemFont(ofSize: fontSize, weight: .regular)
    }

    /// Highlight code with optional language hint
    func highlight(_ code: String, language: String? = nil) -> NSAttributedString {
        // Handle empty code
        guard !code.isEmpty else {
            return NSAttributedString(string: "", attributes: baseAttrs())
        }

        let lang = (language ?? detectLanguage(code)).lowercased()
        let langKeywords = keywords[lang] ?? keywords["python"]!.union(keywords["java"]!).union(keywords["javascript"]!)

        let result = NSMutableAttributedString()
        let lines = code.components(separatedBy: "\n")
        var inMultiLineComment = false

        for (i, line) in lines.enumerated() {
            let (highlighted, stillInComment) = highlightLine(line, keywords: langKeywords, lang: lang, inComment: inMultiLineComment)
            inMultiLineComment = stillInComment
            result.append(highlighted)
            if i < lines.count - 1 {
                result.append(NSAttributedString(string: "\n", attributes: baseAttrs()))
            }
        }

        return result
    }

    /// Get background color for code blocks
    var backgroundColor: NSColor { codeBgColor }

    // MARK: - Private

    private func baseAttrs() -> [NSAttributedString.Key: Any] {
        [.font: codeFont, .foregroundColor: plain]
    }

    private func detectLanguage(_ code: String) -> String {
        let lower = code.lowercased()
        if lower.contains("def ") && lower.contains(":") { return "python" }
        if lower.contains("public class") || lower.contains("public static void") { return "java" }
        if lower.contains("function") || lower.contains("const ") || lower.contains("=>") { return "javascript" }
        if lower.contains("select") && lower.contains("from") { return "sql" }
        if lower.contains("func ") && lower.contains("->") { return "swift" }
        if lower.contains("package ") && lower.contains("func ") { return "go" }
        return "generic"
    }

    private func highlightLine(_ line: String, keywords: Set<String>, lang: String, inComment: Bool) -> (NSAttributedString, Bool) {
        let result = NSMutableAttributedString()
        let stillInComment = inComment

        // Handle empty line
        guard !line.isEmpty else {
            return (result, stillInComment)
        }

        // Handle ongoing multi-line comment
        if stillInComment {
            if let endIdx = line.range(of: "*/") {
                let commentPart = String(line[..<endIdx.upperBound])
                result.append(NSAttributedString(string: commentPart, attributes: [.font: codeFont, .foregroundColor: comment]))
                let rest = String(line[endIdx.upperBound...])
                let (restHighlighted, _) = highlightLine(rest, keywords: keywords, lang: lang, inComment: false)
                result.append(restHighlighted)
                return (result, false)
            } else {
                result.append(NSAttributedString(string: line, attributes: [.font: codeFont, .foregroundColor: comment]))
                return (result, true)
            }
        }

        // Find single-line comment start
        let commentStart = findSingleLineComment(line, lang: lang)

        // Find multi-line comment start
        if let multiStart = line.range(of: "/*") {
            if commentStart == nil || multiStart.lowerBound < commentStart! {
                let before = String(line[..<multiStart.lowerBound])
                result.append(highlightTokens(before, keywords: keywords))
                let after = String(line[multiStart.lowerBound...])
                // Safe range check - need at least 3 chars to search for */ after /*
                if after.count > 2 {
                    let searchStart = after.index(after.startIndex, offsetBy: 2)
                    if let endIdx = after.range(of: "*/", range: searchStart..<after.endIndex) {
                        let commentText = String(after[..<endIdx.upperBound])
                        result.append(NSAttributedString(string: commentText, attributes: [.font: codeFont, .foregroundColor: comment]))
                        let remaining = String(after[endIdx.upperBound...])
                        result.append(highlightTokens(remaining, keywords: keywords))
                        return (result, false)
                    }
                }
                // No closing */ found on this line
                result.append(NSAttributedString(string: after, attributes: [.font: codeFont, .foregroundColor: comment]))
                return (result, true)
            }
        }

        // Handle single-line comment
        if let commentIdx = commentStart {
            let code = String(line[..<commentIdx])
            let commentText = String(line[commentIdx...])
            result.append(highlightTokens(code, keywords: keywords))
            result.append(NSAttributedString(string: commentText, attributes: [.font: codeFont, .foregroundColor: comment]))
            return (result, false)
        }

        // No comments - highlight tokens
        result.append(highlightTokens(line, keywords: keywords))
        return (result, false)
    }

    private func findSingleLineComment(_ line: String, lang: String) -> String.Index? {
        var inString = false
        var stringChar: Character = "\""
        var i = line.startIndex

        while i < line.endIndex {
            let char = line[i]

            if !inString && (char == "\"" || char == "'") {
                inString = true
                stringChar = char
            } else if inString && char == stringChar {
                if i > line.startIndex && line[line.index(before: i)] != "\\" {
                    inString = false
                }
            }

            if !inString {
                if char == "/" && line.index(after: i) < line.endIndex && line[line.index(after: i)] == "/" {
                    return i
                }
                if char == "#" && lang == "python" { return i }
                if char == "-" && lang == "sql" && line.index(after: i) < line.endIndex && line[line.index(after: i)] == "-" {
                    return i
                }
            }
            i = line.index(after: i)
        }
        return nil
    }

    private func highlightTokens(_ text: String, keywords: Set<String>) -> NSAttributedString {
        let result = NSMutableAttributedString()

        // Handle empty text
        guard !text.isEmpty else { return result }

        var i = text.startIndex

        while i < text.endIndex {
            let char = text[i]

            // Strings
            if char == "\"" || char == "'" {
                let (str, end) = extractString(text, from: i, quote: char)
                result.append(NSAttributedString(string: str, attributes: [.font: codeFont, .foregroundColor: string]))
                i = end
                continue
            }

            // Decorators
            if char == "@" {
                let (word, end) = extractWord(text, from: i, includePrefix: true)
                result.append(NSAttributedString(string: word, attributes: [.font: codeFont, .foregroundColor: decorator]))
                i = end
                continue
            }

            // Numbers
            let nextIndex = text.index(after: i)
            let hasNextChar = nextIndex < text.endIndex
            if char.isNumber || (char == "." && hasNextChar && text[nextIndex].isNumber) {
                let (num, end) = extractNumber(text, from: i)
                result.append(NSAttributedString(string: num, attributes: [.font: codeFont, .foregroundColor: number]))
                i = end
                continue
            }

            // Words
            if char.isLetter || char == "_" {
                let (word, end) = extractWord(text, from: i)
                let color: NSColor
                if keywords.contains(word) {
                    color = keyword
                } else if builtinTypes.contains(word) || (word.first?.isUppercase == true && word.count > 1) {
                    color = type
                } else if end < text.endIndex && text[end] == "(" {
                    color = function
                } else {
                    color = plain
                }
                result.append(NSAttributedString(string: word, attributes: [.font: codeFont, .foregroundColor: color]))
                i = end
                continue
            }

            // Default character
            result.append(NSAttributedString(string: String(char), attributes: baseAttrs()))
            i = text.index(after: i)
        }

        return result
    }

    private func extractString(_ text: String, from start: String.Index, quote: Character) -> (String, String.Index) {
        var result = String(quote)
        var i = text.index(after: start)
        while i < text.endIndex {
            let char = text[i]
            result.append(char)
            if char == quote && (i == text.index(after: start) || text[text.index(before: i)] != "\\") {
                return (result, text.index(after: i))
            }
            i = text.index(after: i)
        }
        return (result, text.endIndex)
    }

    private func extractWord(_ text: String, from start: String.Index, includePrefix: Bool = false) -> (String, String.Index) {
        var result = ""
        var i = start
        if includePrefix && i < text.endIndex && text[i] == "@" {
            result.append("@")
            i = text.index(after: i)
        }
        while i < text.endIndex && (text[i].isLetter || text[i].isNumber || text[i] == "_") {
            result.append(text[i])
            i = text.index(after: i)
        }
        return (result, i)
    }

    private func extractNumber(_ text: String, from start: String.Index) -> (String, String.Index) {
        var result = ""
        var i = start
        var hasDecimal = false
        while i < text.endIndex {
            let char = text[i]
            if char.isNumber || char.isHexDigit {
                result.append(char)
            } else if char == "." && !hasDecimal {
                hasDecimal = true
                result.append(char)
            } else if (char == "x" || char == "X") && result == "0" {
                result.append(char)
            } else {
                break
            }
            i = text.index(after: i)
        }
        return (result, i)
    }
}

File: Presentation/Styling/SyntaxHighlighter.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;

namespace InterviewMasterApp.Presentation.Styling
{
    /// <summary>
    /// Syntax highlighter for code blocks using One Dark theme colors (string-based).
    /// Ported from Swift: SyntaxHighlighter.swift
    /// </summary>
    public class SyntaxHighlighter
    {
        // One Dark theme colors
        private readonly Color _keyword = Color.FromArgb(199, 120, 221);
        private readonly Color _type = Color.FromArgb(229, 192, 123);
        private readonly Color _string = Color.FromArgb(152, 195, 121);
        private readonly Color _number = Color.FromArgb(209, 154, 102);
        private readonly Color _comment = Color.FromArgb(229, 192, 76);
        private readonly Color _function = Color.FromArgb(97, 175, 239);
        private readonly Color _plain = Color.FromArgb(223, 229, 234);
        private readonly Color _decorator = Color.FromArgb(229, 192, 123);
        private readonly Color _codeBg = Color.FromArgb(128, 31, 33, 38);

        private readonly float _codeFontSize;

        private readonly Dictionary<string, HashSet<string>> _keywords = new(StringComparer.OrdinalIgnoreCase)
        {
            ["python"] = new HashSet<string>(new [] {"def","class","if","elif","else","for","while","return","import","from","as","try","except","finally","raise","with","yield","lambda","pass","break","continue","and","or","not","in","is","async","await","None","True","False","global","nonlocal","assert","del"}, StringComparer.OrdinalIgnoreCase),
            ["java"] = new HashSet<string>(new [] {"public","private","protected","class","interface","extends","implements","static","final","void","int","long","double","float","boolean","char","byte","short","new","return","if","else","for","while","do","switch","case","default","break","continue","try","catch","finally","throw","throws","import","package","this","super","null","true","false","abstract","synchronized","volatile","instanceof","enum","var","record"}, StringComparer.OrdinalIgnoreCase),
            ["javascript"] = new HashSet<string>(new [] {"function","const","let","var","if","else","for","while","do","switch","case","default","break","continue","return","try","catch","finally","throw","class","extends","new","this","super","import","export","from","as","async","await","yield","of","in","typeof","instanceof","delete","void","null","undefined","true","false","NaN"}, StringComparer.OrdinalIgnoreCase),
            ["sql"] = new HashSet<string>(new [] {"SELECT","FROM","WHERE","JOIN","LEFT","RIGHT","INNER","OUTER","ON","AND","OR","NOT","IN","IS","NULL","LIKE","BETWEEN","GROUP","BY","ORDER","ASC","DESC","LIMIT","OFFSET","HAVING","INSERT","INTO","VALUES","UPDATE","SET","DELETE","CREATE","TABLE","ALTER","DROP","INDEX","VIEW","DISTINCT","AS","CASE","WHEN","THEN","ELSE","END","UNION","ALL","EXISTS","COUNT","SUM","AVG","MAX","MIN","COALESCE","CAST"}, StringComparer.OrdinalIgnoreCase),
            ["swift"] = new HashSet<string>(new [] {"func","class","struct","enum","protocol","extension","var","let","if","else","guard","for","while","repeat","switch","case","default","break","continue","return","try","catch","throw","throws","defer","import","self","Self","super","nil","true","false","public","private","internal","static","final","override","mutating","lazy","weak","async","await"}, StringComparer.OrdinalIgnoreCase),
            ["go"] = new HashSet<string>(new [] {"func","package","import","type","struct","interface","var","const","if","else","for","range","switch","case","default","break","continue","return","go","select","chan","map","make","new","defer","panic","recover","nil","true","false","iota"}, StringComparer.OrdinalIgnoreCase)
        };

        private readonly HashSet<string> _builtinTypes = new(StringComparer.OrdinalIgnoreCase)
        {
            "String","Int","Integer","Long","Double","Float","Boolean","Bool","List","Array","Dict","Dictionary","Map","Set","HashMap","HashSet","ArrayList","Optional","Result","Void","Object","Number","Date","Error","Exception","Promise"
        };

        public SyntaxHighlighter(float fontSize = 14)
        {
            _codeFontSize = fontSize;
        }

        public Color BackgroundColor => _codeBg;

        /// <summary>
        /// Highlight code and return styled runs.
        /// </summary>
        public StyledText Highlight(string code, string? language = null)
        {
            var styled = new StyledText();
            if (string.IsNullOrEmpty(code)) return styled;

            var lang = (language ?? DetectLanguage(code)).ToLowerInvariant();
            var langKeywords = _keywords.TryGetValue(lang, out var set) ? set : _keywords["python"].Union(_keywords["java"]).Union(_keywords["javascript"]).ToHashSet(StringComparer.OrdinalIgnoreCase);

            var lines = code.Split('\n');
            bool inMultiComment = false;

            for (int i = 0; i < lines.Length; i++)
            {
                var line = lines[i];
                var (lineRuns, stillIn) = HighlightLine(line, langKeywords, lang, inMultiComment);
                inMultiComment = stillIn;
                styled.Runs.AddRange(lineRuns);
                if (i < lines.Length - 1)
                {
                    styled.Add("\n", BaseStyle());
                }
            }
            return styled;
        }

        private TextStyle BaseStyle()
        {
            return new TextStyle { Foreground = _plain, Monospace = true, FontSize = _codeFontSize };
        }

        private string DetectLanguage(string code)
        {
            var lower = code.ToLowerInvariant();
            if (lower.Contains("def ") && lower.Contains(":")) return "python";
            if (lower.Contains("public class") || lower.Contains("public static void")) return "java";
            if (lower.Contains("function") || lower.Contains("const ") || lower.Contains("=>")) return "javascript";
            if (lower.Contains("select") && lower.Contains("from")) return "sql";
            if (lower.Contains("func ") && lower.Contains("->")) return "swift";
            if (lower.Contains("package ") && lower.Contains("func ")) return "go";
            return "generic";
        }

        private (List<StyledRun> runs, bool stillInComment) HighlightLine(string line, HashSet<string> keywords, string lang, bool inComment)
        {
            var runs = new List<StyledRun>();
            if (string.IsNullOrEmpty(line)) return (runs, inComment);

            if (inComment)
            {
                var endIdx = line.IndexOf("*/", StringComparison.Ordinal);
                if (endIdx >= 0)
                {
                    var commentPart = line.Substring(0, endIdx + 2);
                    runs.Add(new StyledRun(commentPart, CommentStyle()));
                    var rest = line.Substring(endIdx + 2);
                    runs.AddRange(HighlightTokens(rest, keywords));
                    return (runs, false);
                }
                runs.Add(new StyledRun(line, CommentStyle()));
                return (runs, true);
            }

            var singleIdx = FindSingleLineComment(line, lang);
            var multiIdx = line.IndexOf("/*", StringComparison.Ordinal);
            if (multiIdx >= 0 && (singleIdx < 0 || multiIdx < singleIdx))
            {
                var before = line.Substring(0, multiIdx);
                runs.AddRange(HighlightTokens(before, keywords));
                var after = line.Substring(multiIdx);
                var endIdx = after.IndexOf("*/", StringComparison.Ordinal);
                if (endIdx >= 0)
                {
                    var commentText = after.Substring(0, endIdx + 2);
                    runs.Add(new StyledRun(commentText, CommentStyle()));
                    var remaining = after.Substring(endIdx + 2);
                    runs.AddRange(HighlightTokens(remaining, keywords));
                    return (runs, false);
                }
                runs.Add(new StyledRun(after, CommentStyle()));
                return (runs, true);
            }

            if (singleIdx >= 0)
            {
                var code = line.Substring(0, singleIdx);
                var comment = line.Substring(singleIdx);
                runs.AddRange(HighlightTokens(code, keywords));
                runs.Add(new StyledRun(comment, CommentStyle()));
                return (runs, false);
            }

            runs.AddRange(HighlightTokens(line, keywords));
            return (runs, false);
        }

        private int FindSingleLineComment(string line, string lang)
        {
            bool inString = false;
            char stringChar = '"';
            for (int i = 0; i < line.Length; i++)
            {
                var c = line[i];
                if (!inString && (c == '"' || c == '\'')) { inString = true; stringChar = c; }
                else if (inString && c == stringChar)
                {
                    if (i == 0 || line[i - 1] != '\\') inString = false;
                }

                if (!inString)
                {
                    if (c == '/' && i + 1 < line.Length && line[i + 1] == '/') return i;
                    if (c == '#' && lang == "python") return i;
                    if (c == '-' && lang == "sql" && i + 1 < line.Length && line[i + 1] == '-') return i;
                }
            }
            return -1;
        }

        private IEnumerable<StyledRun> HighlightTokens(string text, HashSet<string> keywords)
        {
            var runs = new List<StyledRun>();
            if (string.IsNullOrEmpty(text)) return runs;

            int i = 0;
            while (i < text.Length)
            {
                var c = text[i];

                // Strings (single/double)
                if (c == '"' || c == '\'')
                {
                    var (str, end) = ExtractString(text, i, c);
                    runs.Add(new StyledRun(str, StringStyle()));
                    i = end;
                    continue;
                }

                // Decorator
                if (c == '@')
                {
                    var (word, end) = ExtractWord(text, i, includePrefix: true);
                    runs.Add(new StyledRun(word, DecoratorStyle()));
                    i = end;
                    continue;
                }

                // Numbers
                if (char.IsDigit(c) || (c == '.' && i + 1 < text.Length && char.IsDigit(text[i + 1])))
                {
                    var (num, end) = ExtractNumber(text, i);
                    runs.Add(new StyledRun(num, NumberStyle()));
                    i = end;
                    continue;
                }

                // Words
                if (char.IsLetter(c) || c == '_')
                {
                    var (word, end) = ExtractWord(text, i, includePrefix: false);
                    TextStyle style;
                    if (keywords.Contains(word)) style = KeywordStyle();
                    else if (_builtinTypes.Contains(word) || (word.Length > 1 && char.IsUpper(word[0]))) style = TypeStyle();
                    else if (end < text.Length && text[end] == '(') style = FunctionStyle();
                    else style = BaseStyle();
                    runs.Add(new StyledRun(word, style));
                    i = end;
                    continue;
                }

                // Default char
                runs.Add(new StyledRun(c.ToString(), BaseStyle()));
                i++;
            }

            return runs;
        }

        private (string, int) ExtractString(string text, int start, char quote)
        {
            var result = new List<char> { quote };
            int i = start + 1;
            while (i < text.Length)
            {
                var c = text[i];
                result.Add(c);
                if (c == quote && (i == start + 1 || text[i - 1] != '\\')) return (new string(result.ToArray()), i + 1);
                i++;
            }
            return (new string(result.ToArray()), text.Length);
        }

        private (string, int) ExtractWord(string text, int start, bool includePrefix)
        {
            int i = start;
            var sb = new List<char>();
            if (includePrefix && i < text.Length && text[i] == '@') { sb.Add('@'); i++; }
            while (i < text.Length && (char.IsLetterOrDigit(text[i]) || text[i] == '_'))
            {
                sb.Add(text[i]);
                i++;
            }
            return (new string(sb.ToArray()), i);
        }

        private (string, int) ExtractNumber(string text, int start)
        {
            int i = start;
            var sb = new List<char>();
            bool hasDecimal = false;
            while (i < text.Length)
            {
                var c = text[i];
                if (char.IsDigit(c)) { sb.Add(c); }
                else if (c == '.' && !hasDecimal) { hasDecimal = true; sb.Add(c); }
                else if ((c == 'x' || c == 'X') && sb.Count == 1 && sb[0] == '0') { sb.Add(c); }
                else break;
                i++;
            }
            return (new string(sb.ToArray()), i);
        }

        private TextStyle KeywordStyle() => new TextStyle { Foreground = _keyword, Monospace = true, FontSize = _codeFontSize };
        private TextStyle TypeStyle() => new TextStyle { Foreground = _type, Monospace = true, FontSize = _codeFontSize };
        private TextStyle StringStyle() => new TextStyle { Foreground = _string, Monospace = true, FontSize = _codeFontSize };
        private TextStyle NumberStyle() => new TextStyle { Foreground = _number, Monospace = true, FontSize = _codeFontSize };
        private TextStyle CommentStyle() => new TextStyle { Foreground = _comment, Monospace = true, FontSize = _codeFontSize };
        private TextStyle FunctionStyle() => new TextStyle { Foreground = _function, Monospace = true, FontSize = _codeFontSize };
        private TextStyle DecoratorStyle() => new TextStyle { Foreground = _decorator, Monospace = true, FontSize = _codeFontSize };
    }
}

